### PR TITLE
ensure that if we fetch the identifier that we have a uuid if the node is referenceable

### PR DESCRIPTION
**Pull request by lsmith77**
help fix https://github.com/jackalope/jackalope-jackrabbit/pull/117

### CODE REVIEW

File: src/Jackalope/Node.php [hunk 1]
```diff
@@ -819,6 +825,42 @@ public function getPrimaryItem()
...
+     */
+    public function setUuidGenerator(\Closure $generator)
+    {
+        $this->uuidGenerator = $generator;
+    }
+
+    /**
+     * @return callable a uuid generator function.
+     */
+    protected function getUuidGenerator()
```
Comment (dbu) [inline src/Jackalope/Node.php]: why do we need the additional flexibility with the uuid generator? and is this the best way we can do it? and now that https://github.com/phpcr/phpcr-utils/pull/159 is merged, do we need this?

i would see this more as a 2.0 topic, to make the UuidHelper non-static but an interface that we inject or get from the factory or however we build these things in 2.0.

would it be enough to drop the special case in the ValueConverter and leave this change out?

File: src/Jackalope/Node.php [hunk 2]
```diff
@@ -827,7 +869,11 @@ public function getIdentifier()
     {
         $this->checkState();
 
-        if (isset($this->properties['jcr:uuid'])) {
+        if ($this->isNodeType('mix:referenceable')) {
+            if (empty($this->properties['jcr:uuid'])) {
+                $this->setProperty('jcr:uuid', $this->generateUuid());
+            }
```
Comment (dbu) [inline src/Jackalope/Node.php]: :+1:

File: src/Jackalope/Node.php (hunk 1 above)
Comment (lsmith77) [inline src/Jackalope/Node.php]: true. i guess we do not need this extension point anymore

Comment (lsmith77) [inline src/Jackalope/Node.php]: done

**Pull request merged**